### PR TITLE
Add tolerance to subset_gridpoint. added distance function

### LINKED ISSUE #178: add optional threshold in utils.subset_gridpoint
* xclim version:
* Python version:
* Operating System:

### Description

Describe what you were trying to get done.
Tell us what happened, what went wrong, and what you expected to happen.

### What I Did

```
Paste the command(s) you ran and the output.
If there was a crash, please include the traceback here.
```

### PR DESCRIPTION
**Pull request by huard**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated

### After merging to master and before closing the branch:
- [ ] bumpversion (minor / major / patch) has been called on `master` branch
- [ ] Tags have been pushed

* **What kind of change does this PR introduce?** 
Adds a distance function (in subset, maybe not the most obvious place). 

* **Does this PR introduce a breaking change?** 
No. 

* **Other information**:
I suggest we update the version to 0.12.0 with this new feature. 

This PR fixes #178

### CODE REVIEW

**Comment by huard** [comment]
@sbiner Est-ce que ça fait ton affaire ? En ce moment on passe la distance en mètres, mais on pourrait passer un string avec les unités. Ton avis ?

**Comment by tlogan2000** [comment]
This looks good.  I am wondering now whether an option of finding all points within a distance could be useful as well...  i.e. options of 'closest_pt' and 'points_within' but working with the 'tolerance' value.  Not critical but maybe something to consider

File: xclim/subset.py [hunk 1]
```diff
@@ -374,23 +377,20 @@ def subset_gridpoint(
...
-                iy, ix = np.unravel_index(np.argmin(dist, axis=None), dist.shape)
-                xydims = [x for x in da.lon.dims]
+                # Calculate the geodesic distance between grid points and the point of interest.
+                dist = distance(da, lon, lat)
+
+                # Find the indices for the closest point
+                iy, ix = np.unravel_index(dist.argmin(), dist.shape)
+
+                # Select data from closest point
+                xydims = [x for x in dist.dims]
```
Comment (sbiner) [inline xclim/subset.py]: Est-ce que le code fonctionnerait si `da` a plus que deux dimensions? Ça ne m'apparait pas clair.

Comment (huard) [reply xclim/subset.py]: Non. En ce moment on cherche seulement les lat-lon seulement. 

Je ne pense pas qu'on supporte le masque non plus.

**Pull request merged**